csrc/device_lower/pass/expr_sort.cpp

```diff
@@ -1749,17 +1749,18 @@ std::vector<Expr*> ExprSegmentationSorter::getExprs() const {
 
 std::vector<Expr*> reorderExprsForComputeAt() {
   auto fusion = FusionGuard::getCurFusion();
+  NVF_ERROR(fusion != nullptr);
+
+  // `Fusion::isNoOp` returns true on some special cases for which
+  // `ExprSegmentationSorter::sort` doesn't return an empty list. Therefore,
+  // this check is needed at this moment.
   if (fusion->isNoOp()) {
     return {};
   }
-  NVF_ERROR(fusion != nullptr);
+
   ExprSegmentationSorter sorter(fusion);
   sorter.sort();
-  auto sorted_exprs = sorter.getExprs();
-  NVF_ERROR(
```

Comment:
`ExprSegmentationSorter` skips generating code for output aliases, so the expression list can be empty.